rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2033,11 +2033,14 @@ extern "C" rmw_publisher_t * rmw_create_publisher(
 
 extern "C" rmw_ret_t rmw_get_gid_for_publisher(const rmw_publisher_t * publisher, rmw_gid_t * gid)
 {
-  RET_NULL(publisher);
-  RET_WRONG_IMPLID(publisher);
-  RET_NULL(gid);
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    publisher,
+    publisher->implementation_identifier,
+    eclipse_cyclonedds_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  RMW_CHECK_ARGUMENT_FOR_NULL(gid, RMW_RET_INVALID_ARGUMENT);
```

Comment:
does the order in the docs matter?
e.g.: https://github.com/ros2/rmw/blob/a680b34da1b5be45d946327ffb3f19386e504af1/rmw/include/rmw/rmw.h#L2110-L2111

if you pass a `rmw_publisher_t` with invalid identifier and a null `gid`, which error should be returned?

I'm fine if that's undefined, I don't think it matters, but I wanted to double check.

Reply:
So far, we've left that bit undefined across the entire API. It seemed like an unnecessary constrain.